compiler/ccgtypes.nim

```diff
@@ -536,7 +541,13 @@ proc genVirtualProcParams(m: BModule; t: PType, rettype, params: var string,
     names.add name
   multiFormat(params, @['\'', '#'], [types, names])
   if params == "()":
-    params = "(void)"
+    if types.len == 0:
+      params = "(void)"
+    else:
+      if not isCtor:
+        names.delete(0)
+        types.delete(0)
+      params = "(" & zip(types, names).mapIt(it[0] & " " & it[1]).join(", ") & ")"
```

Comment:
This line with `zip` + `mapIt` + `join` feels kinda golfed for the compiler core,
can it be like a for loop that appends to a string or something like that?. 🤔